recipe/patches/0100-osx_bigsur_compatibility.diff

```diff
@@ -0,0 +1,22 @@
+diff --git a/qtbase/src/plugins/platforms/cocoa/qnsview_drawing.mm b/qtbase/src/plugins/platforms/cocoa/qnsview_drawing.mm
+index 116a9a7..14e45a0 100644
+--- a/qtbase/src/plugins/platforms/cocoa/qnsview_drawing.mm
++++ b/qtbase/src/plugins/platforms/cocoa/qnsview_drawing.mm
+@@ -95,8 +95,15 @@
+     // by AppKit at a point where we've already set up other parts of the platform plugin
+     // based on the presence of layers or not. Once we've rewritten these parts to support
+     // dynamically picking up layer enablement we can let AppKit do its thing.
+-    return QMacVersion::buildSDK() >= QOperatingSystemVersion::MacOSMojave
+-        && QMacVersion::currentRuntime() >= QOperatingSystemVersion::MacOSMojave;
```

Comment:
@hmaarrfk, this condition will return true if it's on Big Sur. Does it not?

Reply:
Ok, I see what you mean.

Because we already moved from OSX 10.12 to OSX 10.15, the buildSDK is greater than Mojave, which would trigger this condition to be true.

how does "BuildTime" SDK relate to the "targetSDK"? I feel like the new logic officially introduced by Qt 
https://bugreports.qt.io/browse/QTBUG-87014
https://codereview.qt-project.org/c/qt/qtbase/+/322228

Is simply "correct" as opposed to depending specifically on the build time SDK.

The build scripts we updated would allow compiling with 10.13, which is lower than Mojave. I feel like for "safety" it is better to keep the patch in.

Reply:
> how does "BuildTime" SDK relate to the "targetSDK"?

BuildTime SDK needs to be greater or equal to the targetSDK.

> The build scripts we updated would allow compiling with 10.13, which is lower than Mojave. I feel like for "safety" it is better to keep the patch in.

Yes, but we are not compiling with 10.13, but with 10.15, so that's fine. I don't like having patches here which are stricly not necessary.

Reply:
Would it be acceptable then if I asked to raise the minimum build time SDK to 10.14 in the `build.sh` script?

Reply:
I don't understand. Build time SDK is fixed to be 10.15

Reply:
ok.